### ISSUE TITLE
Doesn't work properly with git-lfs enabled repositories

### ISSUE THREAD

**Issue by alanwilter** [microsoft/vscode#146190]
- VS Code Version: 1.65.2
- OS Version: macOS 11.6.5

This issue is an updated version of #24705

My repo uses `git-lfs` and it's installed in conda env.

The VSCode workspace, is set to use the same conda for the python interpreter.

When running git command on VSCode terminal, it all works fine but if I use VSCode graphic interface, to `Discard changes` for a file (`git restore ...` in CLI), it will throw an error:

<img width="259" alt="Screenshot 2022-03-28 at 16 13 40" src="https://user-images.githubusercontent.com/3899850/160430082-2dabc8da-7835-4ff7-ae21-62d41f070ae2.png">

`Show Command Output` gives:

```
This repository is configured for Git LFS but 'git-lfs' was not found on your path.
If you no longer wish to use Git LFS, remove this hook by deleting '.git/hooks/post-checkout'.
```

From `Developer Tools`

```
process.env.PATH
"/usr/bin:/bin:/usr/sbin:/sbin"
```

**Comment by ynyyn** [comment]
I had the same problem on VSCode 1.66.2 on macOS 11.6.1.

In VSCode DevTools:

```js
> process.env.PATH
< '/usr/bin:/bin:/usr/sbin:/sbin'
> process.env.SHELL
< '/bin/zsh'
```

In external Terminal:

```shell
which git-lfs
/usr/local/bin/git-lfs
```

So intuitively it's an environment variables problem that where `git-lfs` located (`/usr/local/bin/`) is not included in VSCode's internal `PATH`.

But, I am sure it was working fine in the past, so I believe there must be something wrong.

One clue is that it seems go wrong after a session recovery coming from system reboot (or a system Crash of WindowServer), and maybe this process have some glitch, I doubt.

So I decided to restart VSCode process..., and it turned out the problem did go away magically as hoped.
But when I checked the `process.env.PATH` in devtools again, to my surprise, there is nothing changed. (still `'/usr/bin:/bin:/usr/sbin:/sbin'`, but `git-lfs` works.)

So to summarize:
- In some special launch case, `git-lfs` may do break due to some glitch. From my own experience, it may have something to do with session recovery caused by system reboot or system UI crash.
- Internal `git-lfs` invocation seems not affected by`process.env.PATH`, there exists some other search mechanism.

**Comment by smallp** [comment]
Same issue. The log of git is `git-lfs filter-process: git-lfs: command not found`. I have to run commands in console.

**Comment by lszomoru** [comment]
@bpasero, I tend to remember seeing this behaviour on MacOS when after a restart, the environment fails to load correctly when VS Code is restored. Are you aware of any issues in this space? Is there a tracked issue for that? Thanks!

**Comment by bpasero** [comment]
We have a timeout and then show an error. The request to make that configurable is https://github.com/microsoft/vscode/issues/132669

**Comment by bpasero** [comment]
I share the concern and actually even today decided to finally get rid of `nvm` on my intel based macOS where I was also constantly seeing this after restart. Now I no longer have this issue and VSCode can instantly be used. I prefer this new behaviour over staying with `nvm` and letting it take 30 seconds (?).

Maybe @joaomoreno or @alexdima can also chime in because I am not entirely sure a setting is the right solution for this. Lets reflect what it means to wait a long time for the call:
* you cannot use any extension because the extension host waits for the environment
* you cannot use terminals because they also wait
* you can actually not do anything that does a network request via `IRequestService`
* the UI opens quite instantly, we e.g. do not wait to open an editor until the environment is resolved

Today, after 10 seconds at least you get feedback via the notification what is going on and have a help document to try to resolve it. If we now go to have a setting for e.g. 30 seconds or 1 minute and you open VS Code there is no feedback at all about the shell environment being resolved. Opening a terminal just hangs and the editor does not have any language features. Outgoing network calls will also stall (I assume for example for extension installation).

Happy to be convinced otherwise if we find a solution that is not frustrating on the other spectrum.

**Comment by lszomoru** [comment]
Yes, I do have `nvm` installed and I am happy to go ahead and uninstall it. The problem that I see with the current flow is that we show an error notification (👍🏻) that only contains a button to learn more yet we do know that there is a great chance that the user will run into some kind of problem given that we weren't able to load the environment (👎🏻). 

I would suggest that we tweak the message, and provide an action to the user to fully restart VS Code with the hope that the issue is transient and on the next start we are able to successfully load the environment 🤞🏻

**Comment by bpasero** [comment]
Given this new progress indicator, maybe we can actually allow to configure the 10s timeout 🤔 , but I want to hear from Alex or Joao first.

**Comment by lszomoru** [comment]
Thanks @bpasero for the change.

**Comment by joaomoreno** [comment]
I like this! Even configuring the timeout, I don't see why not.

This is now possible since we moved the resolution closer to the features that require it. It used not to be possible, since we were doing this on startup and the entire workbench depended on it.

**Comment by bpasero** [comment]
A new setting `application.shellEnvironmentResolutionTimeout` is available beginning tomorrow.

**Comment by lszomoru** [comment]
@bpasero, thank you very much for all these improvements.
 @alanwilter, I will go ahead and close the issue but please feel free to reopen it in case you still have any questions. Thanks!
